File: CodingBatPython/Logic2.py
import math


def make_bricks(small, big, goal):
    return goal % 5 <= small and goal - big * 5 <= small

def lone_sum(a, b, c):
    total = 0
    if a != b and a != c:
        total += a
    if b != a and b != c:
        total += b
    if c != a and c != b:
        total += c
    return total

def lucky_sum(a, b, c):
    if a == 13:
        return 0
    if b == 13:
        return a
    if c == 13:
        return a + b
    return a + b + c

def no_teen_sum(a, b, c):
    return fix_teen(a) + fix_teen(b) + fix_teen(c)


def fix_teen(n):
    if 13 <= n <= 19 and n != 15 and n != 16:
        return 0
    return n

def round_sum(a, b, c):
    return round10(a) + round10(b) + round10(c)

def round10(num):
    if num % 10 < 5:
        return math.trunc(num / 10) * 10
    return math.trunc(num / 10) * 10 + 10

def close_far(a, b, c):
    return (abs(a - b) <= 1 and abs(a - c) >= 2 and abs(b - c) >= 2) or (abs(a - c) <= 1 and abs(a - b) >= 2 and abs(b - c) >= 2)

def make_chocolate(small, big, goal):
    big_bars = math.trunc(goal / 5)
    if big_bars <= big:
        goal -= big_bars * 5
    else:
        goal -= big * 5
    if goal <= small:
        return goal
    return -1

File: CodingBatJava/src/Logic2.java
public class Logic2 {
    public boolean makeBricks(int small, int big, int goal) {
        return goal % 5 <= small && goal - big * 5 <= small;
    }

    public int loneSum(int a, int b, int c) {
        int sum = 0;
        if (a != b && a != c) sum += a;
        if (b != a && b != c) sum += b;
        if (c != a && c != b) sum += c;
        return sum;
    }

    public int luckySum(int a, int b, int c) {
        if (a == 13) return 0;
        if (b == 13) return a;
        if (c == 13) return a + b;
        return a + b + c;
    }

    public int noTeenSum(int a, int b, int c) {
        return fixTeen(a) + fixTeen(b) + fixTeen(c);
    }

    public int fixTeen(int n) {
        if (n >= 13 && n <= 19 && n != 15 && n != 16) return 0;
        return n;
    }

    public int roundSum(int a, int b, int c) {
        return round10(a) + round10(b) + round10(c);
    }

    public int round10(int num) {
        if (num % 10 < 5) return num / 10 * 10;
        else return (num / 10 + 1) * 10;
    }

    public boolean closeFar(int a, int b, int c) {
        return Math.abs(a - b) <= 1 && Math.abs(a - c) >= 2 && Math.abs(b - c) >= 2 || Math.abs(a - c) <= 1 && Math.abs(a - b) >= 2 && Math.abs(b - c) >= 2;
    }

    public int blackjack(int a, int b) {
        if (a > 21 && b > 21) return 0;
        int max = Math.max(a, b);
        if (max <= 21) return max;
        return Math.min(a, b);
    }

    public boolean evenlySpaced(int a, int b, int c) {
        int min = Math.min(a, Math.min(b, c));
        int max = Math.max(a, Math.max(b, c));
        int mid = a + b + c - min - max;
        return max - mid == mid - min;
    }

    public int makeChocolate(int small, int big, int goal) {
        int bigBars = goal / 5;
        if (bigBars <= big) goal -= bigBars * 5;
        else goal -= big * 5;
        if (goal <= small) return goal;
        return -1;
    }
}